src/backends/tensorrt/plan_backend.cc

```diff
@@ -1794,92 +1847,317 @@ PlanBackend::Context::Run(
       continue;
     }
 
-    const InferenceRequest::Input* repr_input;
-    FAIL_ALL_AND_RETURN_IF_ERROR(
-        payload_->requests_, payload_->responses_, metric_reporter_.get(),
-        repr_input_request->ImmutableInput(name, &repr_input),
-        "failed to obtain the input '" + name + "'");
-    // Get the shape of the input. The request has already checked
-    // that the request shape is valid so don't need to do it here.
-    const auto& batch1_shape = repr_input->Shape();
+    if (buffer_is_ragged_[bindex]) {
```

Comment:
Can you pull out some of the large inline code blocks into separate functions so that this function doesn't become unreadable.

Reply:
I think the processing code can be moved into InputCollector and it returns the shape for the batch input and set the value into the provided buffer, because the processing for different batch input kind should be the same across all backends, the differences may just be how to pass the shape and data to the models.

Reply:
If the code is general then moving most/all of it into the collector would be great.